pynest/examples/pynest-microcircuit/helpers.py

```diff
@@ -0,0 +1,142 @@
+    """This function returns the total number of synapses between
+    all populations.
+
+    The first index (rows) of the matrix is the target population
+    and the second (columns) the source population.
+    This function depends on the following parameters specified in net_dict:
+    number of populations, Number of Neurons (N_full),
+    connection probability (conn_probs), scaling factor (N_scaling)
+
+    Returns: Matrix with total number of synapses,
+        dimension: (len(populations), len(populations)).
+    """
+    N_full = net_dict['N_full']
+    conn_probs = net_dict['conn_probs']
+    scaling = net_dict['N_scaling']
+    prod = np.outer(N_full, N_full)
+    n_syn_temp = np.log(1. - conn_probs)/np.log((prod - 1.) / prod)
+    N_full_matrix = np.column_stack((N_full for i in range(len(N_full))))
+    # if the network is scaled the indegrees are calculated in the same
+    # fashion as in the original version of the circuit, which is
+    # written in sli
+    K = (((n_syn_temp * (
+        N_full_matrix * scaling).astype(int)) / N_full_matrix).astype(int))
+    return K
+
+
+def synapses_th_matrix():
+    """This function returns the total number of synapses,
+        between the thalamus and all target populations.
+
+    Parameters: length of populations, number of neurons (N_full), connection
+        probability (conn_probs), scaling factor (N_scaling)
+
+    Returns: Vector with total number of synapses.
+    """
+    scaling = net_dict['N_scaling']
+    N_full = net_dict['N_full']
+    conn_probs = stim_dict['conn_probs_th']
+    scaling = net_dict['N_scaling']
+    prod = (stim_dict['n_thal'] * net_dict['N_full']).astype(float)
+    n_syn_temp = np.log(1. - conn_probs)/np.log((prod - 1.)/prod)
+    K = (((n_syn_temp * (N_full * scaling).astype(int))/N_full).astype(int))
+    return K
+
+
+def adj_w_ext_to_K(K_full, K_scaling, w, w_from_PSP, DC):
+    """With this funtion the recurrent and external weights are adjusted
+    to the scaling of the indegrees. Extra DC input is added to
+    compensate the scaling and preserve the mean and variance of the input.
+
+    Parameters:
+    Time constant of the external postsynaptic excitatory current (tau_syn_E),
+    mean rates of the populations in the non-scaled version (full_mean_rates),
+    weight calcuted with function 'create_weight_from_PSP' (w_from_PSP),
+    number of external connections to the different populations (K_ext),
+    rate of the poissonian spike generator (bg_rate),
+    scaling factor for the connections (K_scaling),
+    weight matrix, created by the function 'weight_mat' (w),
+    DC input, vector with zeros (DC).
+
+    Returns: Weight matrix, weight for external input, extra DC input.
+    """
+    tau_syn_E = net_dict['n_params']['tau_syn_E']
+    full_mean_rates = net_dict['full_mean_rates']
+    w_mean = w_from_PSP
+    K_ext = net_dict['K_ext']
+    bg_rate = net_dict['bg_rate']
+    internal_scaling = K_scaling
```

Comment:
It seems that the distinction between `internal_scaling` and `external_scaling` is superfluous here.
You can just use `K_scaling` without creating two new variables.